libraries/AP_BattMonitor/AP_BattMonitor.h

```diff
@@ -58,10 +58,10 @@ class AP_BattMonitor
 public:
 
     // battery failsafes must be defined in levels of severity so that vehicles wont fall backwards
-    enum BatteryFailsafe {
-        BatteryFailsafe_None = 0,
-        BatteryFailsafe_Low,
-        BatteryFailsafe_Critical
+    enum class Failsafe {
```

Comment:
needs uint8_t type, Peter told me so

Reply:
Don't listen to him.

Reply:
Needed because without it the build on plane grows by 92 bytes, and the per battery state grows by 4 bytes which is actually a very silly thing to spend 36 bytes of RAM on :)